lib/ts/userArgumentUtils.ts

```diff
@@ -78,6 +81,18 @@ export function validateUserArguments(userArguments: UserFlagsRaw) {
             throw new Error("Invalid package manager provided, valid values:\n" + availableManagers);
         }
     }
+
+    if (userArguments.ui !== undefined) {
+        if (!isValidUIBuildType(userArguments)) {
+            throw new Error(
+                `Invalid argument for ui provided. Supported arguments are ${Object.values(UIBuildType).join(", ")}`
+            );
+        }
+
+        if (!isValidUiBasedOnFilters(userArguments)) {
+            throw new Error(`UI type ${userArguments.ui} is not supported for the selected frontend or recipe.`);
+        }
```

Comment:
we should move this into isValidUIBuildType

Reply:
The code is updated with the requested changes, hence resolving this comment. I have moved the checks into a single function